### PR TITLE
[fix] move all platform dependent code to xfuser/envs.py

### PR DESCRIPTION
**Pull request by houchen-li**
# Description

The source code of xDiT contains few problems:

1. In file `xfuser/core/long_ctx_attention/ring/ring_flash_attn.py:5` , `flex_attention` was imported but never be used. Since this module is introduced in PyTorch 2.5， adding this unused module may prevent xDiT from applying to the models which requires PyTorch < 2.5.

2. In file `xfuser/core/long_ctx_attention/ring/ring_flash_attn.py:108` , the implicit default `softmax_scale` was calculated via `q.shape[-1] ** (-0.5)`, which truncates to an integer number that brokes the entire workflow. The right way to calculate this value is by `1.0 / math.sqrt(q.size(-1))` which ends up with a float-point number.

3. In `xfuser/core/distributed/parallel_state.py`, [previous patch](https://github.com/xdit-project/xDiT/pull/539) involves platform-dependent codes with try-except blocks. This can be improved by moving all platform-dependent codes to `xfuser/envs.py`.

# Features

* remove unused module `flex_attention` from `xfuser/core/long_ctx_attention/ring/ring_flash_attn.py`.
* get rid of the type trunction error when calculate implicit default `softmax_scale`.
* move platform dependent codes to `xfuser/envs.py`

# Risks

* xDiT is now available on PyTorch < 2.5 environment. But I haven't test lots of those models. The stability and accuracy may not be guaranteed.

### CODE REVIEW

**Comment by houchen-li** [comment]
@feifeibear

**Review by feifeibear — APPROVED** [review]
